backend/leveldb/leveldb.go

```diff
@@ -0,0 +1,193 @@
+package leveldb
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"errors"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/syndtr/goleveldb/leveldb"
+	"github.com/syndtr/goleveldb/leveldb/opt"
+	"github.com/syndtr/goleveldb/leveldb/util"
+
+	"github.com/kellegous/go/internal"
+)
+
+const (
+	routesDbFilename = "routes.db"
+	idLogFilename    = "id"
+)
+
+// LevelDBBackend provides access to the leveldb store.
+type LevelDBBackend struct {
```

Comment:
To avoid stuttered terms in the names, we should name this `Backend`. That way we have `leveldb.Backend` instead of `leveldb.LevelDBBackend`.